autonomi/src/networking/driver/task_handler.rs

```diff
@@ -0,0 +1,347 @@
+                trace!(
+                    "QueryId({id}): GetClosestPeersError::Timeout {:?}, peers: {:?}",
+                    hex::encode(key),
+                    peers
+                );
+                responder
+                    .send(Err(NetworkError::GetClosestPeersTimeout))
+                    .map_err(|_| TaskHandlerError::NetworkClientDropped)?;
+            }
+        }
+        Ok(())
+    }
+
+    pub fn update_get_record(
+        &mut self,
+        id: QueryId,
+        res: Result<kad::GetRecordOk, kad::GetRecordError>,
+    ) -> Result<(), TaskHandlerError> {
+        match res {
+            Ok(kad::GetRecordOk::FoundRecord(record)) => {
+                trace!(
+                    "QueryId({id}): GetRecordOk::FoundRecord {:?}",
+                    PrettyPrintRecordKey::from(&record.record.key)
+                );
+                let holders = self.get_record_accumulator.entry(id).or_default();
+                if let Some(peer_id) = record.peer {
+                    holders.insert(peer_id, record.record);
+                }
+            }
+            Ok(kad::GetRecordOk::FinishedWithNoAdditionalRecord { .. }) => {
+                trace!("QueryId({id}): GetRecordOk::FinishedWithNoAdditionalRecord");
+                let (responder, holders) = self.take_responder_and_holders_for_task(id)?;
+                let peers = holders.keys().cloned().collect();
+                let records_uniq = holders.values().cloned().fold(Vec::new(), |mut acc, x| {
+                    if !acc.contains(&x) {
+                        acc.push(x);
+                    }
+                    acc
+                });
+
+                let res = match &records_uniq[..] {
+                    [] => responder.send(Ok((None, peers))),
+                    [one] => responder.send(Ok((Some(one.clone()), peers))),
+                    [_one, _two, ..] => responder.send(Err(NetworkError::SplitRecord(holders))),
+                };
+                res.map_err(|_| TaskHandlerError::NetworkClientDropped)?;
+            }
+            Err(kad::GetRecordError::NotFound { key, closest_peers }) => {
+                trace!(
+                    "QueryId({id}): GetRecordError::NotFound {:?}, closest_peers: {:?}",
+                    hex::encode(key),
+                    closest_peers
+                );
+                let (responder, holders) = self.take_responder_and_holders_for_task(id)?;
+                let peers = holders.keys().cloned().collect();
+
+                responder
+                    .send(Ok((None, peers)))
+                    .map_err(|_| TaskHandlerError::NetworkClientDropped)?;
+            }
+            Err(kad::GetRecordError::QuorumFailed {
+                key,
+                records,
+                quorum,
+            }) => {
+                trace!(
+                    "QueryId({id}): GetRecordError::QuorumFailed {:?}, records: {:?}, quorum: {:?}",
```

Comment:
For mutable datatypes, QuorumFailed scenario could happen with multiple versions exists.
Hence, such fetched records shall be `aggregated`.
here, at least shall report back such multiple versions.